packages/trader/src/Modules/Trading/Components/Form/ContractType/contract-type-widget.tsx

```diff
@@ -33,14 +41,14 @@ const ContractTypeWidget = ({ is_equal, name, value, list, onChange, languageCha
         };
     }, [handleClickOutside]);
 
-    const handleCategoryClick = ({ key }) => {
-        setSelectedCategory(key);
+    const handleCategoryClick: React.ComponentProps<typeof ContractType.Dialog>['onCategoryClick'] = ({ key }) => {
+        if (key) setSelectedCategory(key);
     };
 
-    const handleSelect = (clicked_item, e) => {
+    const handleSelect = (clicked_item: TContractType | undefined, e: React.MouseEvent) => {
```

Comment:
```suggestion
    const handleSelect = (clicked_item: TContractType | undefined, e: React.MouseEvent<HTMLButtonElement | HTMLInputElement>) => {
```

Reply:
do we want to handicap to button and input elements only?

Reply:
yes, we know it's passed to `List` -> `Item` (where div is clicked) and to `Info` -> `Button` (where button is clicked) :)

Reply:
resolved.